sdks/python/apache_beam/runners/sdf_utils.py

```diff
@@ -0,0 +1,173 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pytype: skip-file
+
+"""Common utility class to help SDK harness to execute an SDF. """
+
+from __future__ import absolute_import
+from __future__ import division
+
+import logging
+import threading
+from builtins import object
+from collections import namedtuple
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Optional
+from typing import Tuple
+
+from apache_beam.utils import timestamp
+
+if TYPE_CHECKING:
+  from apache_beam.io.iobase import RestrictionTracker
+  from apache_beam.utils.timestamp import Timestamp
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SplitResultPrimary = namedtuple(
+    'SplitResultPrimary', 'windowed_value')
+
+SplitResultResidual = namedtuple(
+    'SplitResultResidual',
+    'windowed_value current_watermark deferred_timestamp')
+
+class ThreadsafeRestrictionTracker(object):
```

Comment:
I know that this class is just being copied over from another file, but I'd love to see some type annotations added to it.

Reply:
Thanks for pointing this out, but I don't think adding type annotations should be in the scope of this PR. The plan is, after this PR checked in, I'll modify https://github.com/apache/beam/pull/10375 and checked in. Then have another PR to do the type stuff.
And I'm curious how can we test the typing? Do we run `tox -e mypy`?

Reply:
sounds good. 

> And I'm curious how can we test the typing? Do we run tox -e mypy?

`tox -e py37-mypy`

Note that there's an issue with the tox config that's preventing colored output in mypy.  So you may want to do (from a python3 venv):

```
pip install mypy
cd sdks/python
mypy
```

There's also a fair bit of mypy error noise still since not all the PRs have been merged yet, so you'll want to focus on file paths relevant to you change.

Reply:
I agree that adding types would be nice, but is probably out of scope. (It would make sense to add types on the base class at the same time.)

Reply:
Create a JIRA here: https://issues.apache.org/jira/browse/BEAM-9296. I'll follow up when finishing API changes.